sklearn/neighbors/_nearest_centroid.py

```diff
@@ -205,29 +254,32 @@ def fit(self, X, y):
                     )
                 self.centroids_[cur_class] = X[center_mask].mean(axis=0)
 
+        # Compute within-class std_dev with unshrunked centroids
+        variance = np.square(X - self.centroids_[y_ind])
```

Comment:
```suggestion
        variance = (X - self.centroids_[y_ind]) ** 2
```

Reply:
I recall encountering a bug (5 months ago, so memory may be blurry) when using Python's built-in power operator in lieu of numpy's power operator due to incompatibility with array-like data types? I can check again.

Reply:
Everything are NumPy arrays here so it should work.